it/controllers/benefits/reimbursed/OtherBenefitsControllerISpec.scala

```diff
@@ -0,0 +1,447 @@
+    Seq(UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN, Some(ExpectedIndividualEN)),
+      UserScenario(isWelsh = false, isAgent = true, CommonExpectedEN, Some(ExpectedAgentEN)),
+      UserScenario(isWelsh = true, isAgent = false, CommonExpectedCY, Some(ExpectedIndividualCY)),
+      UserScenario(isWelsh = true, isAgent = true, CommonExpectedCY, Some(ExpectedAgentCY)))
+  }
+
+  ".show" should {
+    import Selectors._
+
+    userScenarios.foreach { user =>
+
+      import user.commonExpectedResults._
+
+      s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+        "render 'other benefits' page with the correct content with no pre-filling" which {
+          lazy val result: WSResponse = {
+            dropEmploymentDB()
+            userDataStub(userData(fullEmploymentsModel()), nino, taxYearEOY)
+            insertCyaData(employmentUserData(isPrior = true, cyaModel(benefits = Some(benefits(fullReimbursedCostsVouchersAndNonCashModel.copy(otherItemsQuestion = None))))), userRequest)
+            authoriseAgentOrIndividual(user.isAgent)
+            urlGet(pageUrl(taxYearEOY), welsh = user.isWelsh,
+              headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+
+          "has an OK status" in {
+            result.status shouldBe OK
+          }
+
+          titleCheck(user.specificExpectedResults.get.expectedTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          textOnPageCheck(expectedCaption, captionSelector)
+          radioButtonCheck(yesText, 1, checked = false)
+          radioButtonCheck(noText, 2, checked = false)
+          buttonCheck(expectedButtonText, continueButtonSelector)
+          formPostLinkCheck(continueLink, continueButtonFormSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "render 'other benefits' page with the correct content with yes pre-filled" which {
+          lazy val result: WSResponse = {
+            dropEmploymentDB()
+            userDataStub(userData(fullEmploymentsModel()), nino, taxYearEOY)
+            insertCyaData(employmentUserData(isPrior = true, cyaModel(benefits = Some(benefits(fullReimbursedCostsVouchersAndNonCashModel)))), userRequest)
+            authoriseAgentOrIndividual(user.isAgent)
+            urlGet(pageUrl(taxYearEOY), welsh = user.isWelsh,
+              headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an OK status" in {
+            result.status shouldBe OK
+          }
+
+          titleCheck(user.specificExpectedResults.get.expectedTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          textOnPageCheck(expectedCaption, captionSelector)
```

Comment:
captionCheck can be used here instead of textOnPageCheck

Reply:
updated